spotify_player/src/config/mod.rs

```diff
@@ -227,14 +227,38 @@ impl Default for DeviceConfig {
 }
 
 impl AppConfig {
+    pub fn new(path: &Path, theme: Option<&String>) -> Result<Self> {
+        let mut config = Self::default();
+        if config.parse_config_file(path)?.is_none() {
+            config.write_config_file(path)?
+        }
+
+        if let Some(theme) = theme {
+            config.theme = theme.to_owned()
+        }
+
+        Ok(config)
+    }
+
     // parses configurations from an application config file in `path` folder,
     // then updates the current configurations accordingly.
-    pub fn parse_config_file(&mut self, path: &Path) -> Result<()> {
+    fn parse_config_file(&mut self, path: &Path) -> Result<Option<()>> {
```

Comment:
I would prefer returning `Result<bool>` and the function document should be updated with: "Returns false if no config file found and true otherwise".